components/datetime/src/options/components.rs

```diff
@@ -140,3 +340,52 @@ pub enum TimeZoneName {
     #[cfg_attr(feature = "serde", serde(rename = "short"))]
     Short,
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    // Shorten these for terser tests.
+    type FS = FieldSymbol;
+    type FL = FieldLength;
+
+    #[test]
+    fn test_component_bag_to_vec_field() {
+        let bag = Bag::default();
+        assert_eq!(
+            bag.to_vec_fields(),
+            vec![
+                (FS::Year(fields::Year::Calendar), FL::One).into(),
+                (FS::Month(fields::Month::Format), FL::Wide).into(),
+                (FS::Day(fields::Day::DayOfMonth), FL::One).into(),
+                (FS::Hour(fields::Hour::H24), FL::One).into(),
+                (FS::Minute, FL::One).into(),
+                (FS::Second(fields::Second::Second), FL::One).into(),
+            ]
+        );
+    }
+
+    #[test]
+    fn test_component_bag_to_vec_field2() {
+        let bag = Bag {
+            era: None,
+            year: Some(Numeric::Numeric),
+            month: Some(Month::TwoDigit),
+            day: Some(Numeric::Numeric),
+            weekday: None,
+            hour: None,
+            minute: None,
+            second: None,
+            time_zone_name: None,
+            preferences: None,
+        };
```

Comment:
I agree, and was thinking I need to fix this. I filled a follow-up since it'll touch multiple places across the code.

https://github.com/unicode-org/icu4x/issues/623